### PR TITLE
Validate `op_type` for `_create`

### PR DESCRIPTION
**Pull request by olcbean**
Currently any `op_type` passed to `_create` is simply set to `create` without any validation.

This PR will validate the passed `op_type` to `_create` as discussed in [comment](https://github.com/elastic/elasticsearch/pull/27346#discussion_r150602483)

CC @javanna

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java [hunk 1]
```diff
@@ -63,9 +64,16 @@ public String getName() {
 
         @Override
         public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient client) throws IOException {
+            validateOpType(request.params().get("op_type"));
             request.params().put("op_type", "create");
             return RestIndexAction.this.prepareRequest(request, client);
         }
+
+        public void validateOpType(String opType) {
+            if (null != opType && !"create".equals(opType.toLowerCase(Locale.US))) {
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java]: please use the root locale

Comment (s1monw) [reply core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java]: also please use `== false` for comparison

**Review by s1monw — CHANGES_REQUESTED** [review]

**Comment by s1monw** [comment]
@elasticmachine ok to test

File: core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java (hunk 1 above)
Comment (olcbean) [reply core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java]: @s1monw I just made the requested changes.

File: core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java [hunk 2]
```diff
@@ -63,9 +64,16 @@ public String getName() {
 
         @Override
         public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient client) throws IOException {
+            validateOpType(request.params().get("op_type"));
             request.params().put("op_type", "create");
             return RestIndexAction.this.prepareRequest(request, client);
         }
+
+        public void validateOpType(String opType) {
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java]: maybe make this static and pkg private?

**Review by s1monw — APPROVED** [review]
left one suggestion, can you fix that then I will merge it

File: core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java (hunk 2 above)
Comment (olcbean) [reply core/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java]: @s1monw what about private (or pkg private) and final?
It cannot be set to `static` as the inner class `CreateHandler` itself is not static.

**Comment by s1monw** [comment]
merged thanks

**Pull request merged**